stdlib/@tests/test_cases/builtins/check_slice.py

```diff
@@ -0,0 +1,116 @@
+"""
+- Type hint `slice` should be compatible with the **"all-slices"**:
+    - `slice(None)`, `slice(None, None)` and `slice(None, None, None)`. (⟿ `slice[?, ?, ?]`)
+- Type hint `slice[T]` should be compatible with:
+    - `slice(None)`, `slice(None, None)` and `slice(None, None, None)` (⟿ `slice[?, ?, ?]`)
+    - `slice(t)`, `slice(None, t)` and `slice(None, t, None)`.  (⟿ `slice[?, T, ?]`)
+    - `slice(t, None)` and `slice(t, None, None)`.  (⟿ `slice[T, ?, ?]`)
+    - `slice(t, t)` and `slice(t, t, None)`.  (⟿ `slice[T, T, ?]`)
+- Type hint `slice[X, Y]` should be compatible with:
+    - `slice(None)`, `slice(None, None)` and `slice(None, None, None)` (⟿ `slice[?, ?, ?]`)
+    - `slice(y)`, `slice(None, y)` and `slice(None, y, None)`.  (⟿ `slice[?, Y, ?]`)
+    - `slice(x, None)` and `slice(x, None, None)` (⟿ `slice[X, ?, ?]`)
+    - `slice(x, y)` and `slice(x, y, None)`.  (⟿ `slice[X, Y, ?]`)
+-  Type hint `slice[X, Y, Z]` should be compatible with:
+    - `slice(None)`, `slice(None, None)` and `slice(None, None, None)`. (⟿ `slice[?, ?, ?]`)
+    - `slice(y)`, `slice(None, y)` and `slice(None, y, None)`.   (⟿ `slice[?, Y, ?]`)
+    - `slice(x, None)` and `slice(x, None, None)` (⟿ `slice[X, ?, ?]`)
+    - `slice(x, y)` and `slice(x, y, None)`.  (⟿ `slice[X, Y, ?]`)
+    - `slice(None, None, z)`  (⟿ `slice[?, ?, Z]`)
+    - `slice(None, y, z)`   (⟿ `slice[?, Y, Z]`)
+    - `slice(x, None, z)`   (⟿ `slice[X, ?, Z]`)
+    - `slice(x, y, z)`  (⟿ `slice[X, Y, Z]`)
+"""
+
+from typing import Any
+from typing_extensions import assert_type
+
+# region check slice constructors ------------------------------------------------------
+assert_type(slice(None), "slice[Any, Any, Any]")
+assert_type(slice(None, None), "slice[Any, Any, Any]")
+assert_type(slice(None, None, None), "slice[Any, Any, Any]")
+
+assert_type(slice(1), "slice[Any, int, Any]")
+assert_type(slice(None, 1), "slice[Any, int, Any]")
+assert_type(slice(None, 1, None), "slice[Any, int, Any]")
+
+assert_type(slice(1, None), "slice[int, Any, Any]")
+assert_type(slice(1, None, None), "slice[int, Any, Any]")
+
+assert_type(slice(1, 1), "slice[int, int, Any]")
+assert_type(slice(1, 1, None), "slice[int, int, Any]")
+
+assert_type(slice(1, 1, 1), "slice[int, int, int]")
```

Comment:
strictly speaking, you probably don't need to quote these annotations. Most type checkers can't distinguish generic-in-stub-but-not-runtime classes from generic-in-stub-and-runtime classes.

However, the aim of these test cases is to be as similar as possible to real Python code, so I'm okay with keeping the quotes.

Reply:
Maybe we can get python to support subscripting slice in 3.14. Seems like there is no open issue yet for it though.